packages/realm/src/app-services/User.ts

```diff
@@ -268,11 +268,11 @@ export class User<
       serviceName,
     );
 
-    const { body, ok, status, statusText } = await network.fetch(request);
-    assert(ok, `Request failed: ${statusText} (${status})`);
-    assert(body, "Expected a body in the response");
+    const response = await network.fetch(request);
+    assert(response.ok, () => `Request failed: ${response.statusText} (${response.status})`);
```

Comment:
Could you fill me in on why the 2nd arg needs to be a function here?

Reply:
It doesn't actually need to, but I figured it made sense to compute the error message only when the assertion has actually failed. In general - this overload can be used to avoid generating the error message if the assertion doesn't fail.

Reply:
Thanks for explanation 👍